core/templates/dev/head/pages/exploration_editor/RouterServices.js

```diff
@@ -44,14 +44,20 @@ oppia.factory('routerService', [
       return $location.path();
     }, function(newPath, oldPath) {
       if (newPath === '') {
-        $location.path(oldPath);
+        if (oldPath === '/') {
+          $window.history.go(-1);
+        } else {
+          $location.path(oldPath);
+        }
         return;
       }
-
       if (!oldPath) {
         // This can happen when clicking on links whose href is "#".
         return;
       }
+      if (newPath === '/' && oldPath.includes('/gui/')) {
```

Comment:
We can create an exploration from any page such as the library, get_started etc pages as the navbar has the create button which is fixed. If we do a replaceState in the dashboard page then for other pages the back button will again enter into the loop. 
And I didn't understand what context is being carried.